### PR TITLE
Run macos tests on osx-64 and osx-arm64 runners

### PR DESCRIPTION
**Pull request by dbast**
### Description

Finally available for OSS projects: [GitHub Actions: Introducing the new M1 macOS runner available to open source!](
https://github.blog/changelog/2024-01-30-github-actions-introducing-the-new-m1-macos-runner-available-to-open-source/)

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by dbast** [comment]
yay... the macos-14 runner are better available and seem to be really faster ... but some tests are failing.

File: news/13617-enable-osx-arm64-tests [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+* <news item>
+
+### Docs
+
+* <news item>
+
+### Other
+
+* Run Github Test workflow also on osx-arm64 aka Apple Silicon Runners. Enable
+  osx-arm64 canary builds. Fix / disable broken tests.
```
Comment (beeankha) [inline news/13617-enable-osx-arm64-tests:20]:
```suggestion
* Run Github Test workflow also on `osx-arm64` aka Apple Silicon Runners. Enable
  `osx-arm64` canary builds. Fix / disable broken tests. (#13617)
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -488,7 +488,7 @@ jobs:
           Path(environ["GITHUB_ENV"]).write_text(f"ANACONDA_ORG_LABEL={label}")
 
       - name: Create and upload canary build
-        uses: conda/actions/canary-release@v23.10.0
+        uses: conda/actions/canary-release@main
```
Comment (jezdez) [inline .github/workflows/tests.yml:491]: Let's do a tagged release first (I can do that later today)

**Review by jezdez — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (jezdez) [reply .github/workflows/tests.yml:491]:
```suggestion
        uses: conda/actions/canary-release@v24.2.0
```

**Review by jezdez — APPROVED** [review]

**Pull request merged**